concepts/observability-configurable-logging/08-revised-api.md

```diff
@@ -0,0 +1,290 @@
+-> Add dedotting support as a feature to the `LogPipeline`
+
+### Meaningfulness of LogPreset
+
+It turned out that an OpenSearch integration via a HTTP plugin required for an internal SAP scenario can be much simpler then expected. A real templating is not needed and probably a plain simple pipeline config will be better suited. Also, what presets to provide, we cannot think of other scenarios which can be delivered. The binding to a secret can be done already in the pipeline itself, workload selectors could be added as well
+-> do we need the LogPreset concept?
+
+### Support of additional inputs
+It turned out that there are typical inputs available which you would like to process as well, mainly the kubelet logs
+-> add a source selector
+
+## Revised API
+
+More and more oppinion/abstraction was added to the `LogPipeline` concept and the meaningfullnes of the `LogPreset` is in the room. We could now try to push down the `LogPipeline` to be more native and accepting that the user can do really stupid things in combination of supported plugins and revise the LogPreset concept.
+Or we push up even more the `LogPipeline` API by introducing the convinience of the LogPreset while keeping native support as a customization option clearly indicating an unsupported scenario.
+This proposal favors the second idea by dropping the LogPreset fully and having only one layer which is already an abstraction. Giving no chance to configure fluent-bit natively. For that you can simply go with a custom installation.
+
+```YAML
+kind: LogPipeline
+apiVersion: telemetry.kyma-project.io/v1alpha1
+metadata:
+  name: OpenSearchHTTP-App
+spec:
+  input:
+    application: # maps to central tail pipeline
+      namespaces: [] # maps to rewrite_tag rule
+      excludeNamespaces: []
+      pods: []
+      excludePods: []
+      containers: []
+      excludeContainers: []
+      matchLabels:
+        app: icke
+    system: {} #maps to systemd based input like kubelet logs
+    custom: | # entering unsupported mode
+      Name dummy
+      Dummy {"message":"dummy"}
+  filters:
+    - multilineParser:
+        type: "java"
+    - parser:
+        type: "json" # support of built-in parsers
+    - modify:
+        add:
+          key: cluster_identifier
+          value: ${KUBERNETES_SERVICE_HOST}
+    - modify:
+        remove:
+          key: cluster_identifier
+    - select:
+        include:
+          key:
+          regexp:
+        exclude:
+          key:
+          regexp:
+    - custom: | # no "Match" available, entering unsupported mode
+        Name    record_modifier
+        Record  cluster_identifier ${KUBERNETES_SERVICE_HOST}
+
+  output: #only one output
```

Comment:
Just a though can we can may be have individual crds for each output we would support. With this we can introduce custom outputs which would be wrapper around basic outputs (like http). I find `dedot` under http a bit misleading.

Reply:
I don't get the idea with the wrapping. So you suggest a new CRD `Output` and in the pipeline you just reference a concrete resource of that type which is already filled with concrete parameters or still generic?

Can you just outline quickly an example to get the idea more easy

Reply:
```
kind: http-special
apiVersion: outpuputs.telemetry.kyma-project.io/v1alpha1
metadata:
  name: myhttp
spec:
     dedot: true
      Host:
        value: "icke.com"
      HTTP_User:
        value: "icke"
        fromSecretKeyRef:
          name: my-elastic-credentials
          namespace: default
          key: ES_ENDPOINT
      HTTP_Password:
        fromSecretKeyRef:
          name: my-elastic-credentials
          namespace: default
          key: ES_ENDPOINT
---
  kind: http
apiVersion: outpuputs.telemetry.kyma-project.io/v1alpha1
metadata:
  name: myhttp
spec:
      Host:
        value: "icke.com"
      HTTP_User:
        value: "icke"
        fromSecretKeyRef:
          name: my-elastic-credentials
          namespace: default
          key: ES_ENDPOINT
      HTTP_Password:
        fromSecretKeyRef:
          name: my-elastic-credentials
          namespace: default
          key: ES_ENDPOINT
```

Reply:
We agreed to not extract the output into a dedicated CRD as we don't want to model the SAP specific scenario of a OpenSearchHTTP output explicitly but stay with common http plugin. Also, a re-use scenario we cannot foresee for now.